guides/plugins/plugins/storefront/add-custom-controller.md

```diff
@@ -1,2 +1,216 @@
-# Add custom controller
+# Add custom storefront controller
+
+## Overview
+
+In this guide you'll learn how to create a custom storefront controller.
+
+## Prerequisites
+
+In order to add your own controller for your plugin, you first need a plugin as base.
+Therefore, you can refer to the [Plugin Base Guide](../plugin-base-guide.md).
+
+## Adding custom storefront controller
+
+### Storefront Controller class example
+
+First of all we have to create a new controller which extends from the `StorefrontController` class.
+A controller is also just a service which can be registered via the DI-Container.
```

Comment:
```suggestion
A controller is also just a service which can be registered via the service container.
```
( Symfony documents their container as service container instead of DI-container )